src/variable.cpp

```diff
@@ -97,6 +97,47 @@ template <class T> struct RebinHelper {
   }
 };
 
+template <typename T> struct RebinGeneralHelper {
+  static void rebin(const Dim dim, const Variable &oldT,
+                         Variable &newT,
+                         const Variable &oldCoordT,
+                         const Variable &newCoordT) {
+    const auto oldSize = oldT.dimensions()[dim];
+    const auto newSize = newT.dimensions()[dim];
+
+    const auto *xold = oldCoordT.span<T>().data();
+    const auto *xnew = newCoordT.span<T>().data();
+    // This function assumes that dimensions between coord and data either
+    // match, or coord is 1D.
```

Comment:
This comment is outdated (copied from the other implementation). Here it is assumed to be 1D.